service-providers/ip-packet-router/src/lib.rs

```diff
@@ -270,6 +296,47 @@ impl IpPacketRouter {
     }
 }
 
+fn parse_packet(
+    packet: &[u8],
+) -> Result<(&str, IpAddr, IpAddr, Option<SocketAddr>), IpPacketRouterError> {
```

Comment:
given your return type has 4 different things in it (and both being the same type), maybe you should wrap it on some struct instead?